### PR TITLE
Add SourceAddr and TLS metadata to lj.Batch

### PR DESCRIPTION
**Pull request by andrewkroh**
Add connection metadata to `lj.Batch`. This provides the source address
of the connection and TLS metadata (like peer certification for mTLS).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Added
 
+- Add connection metadata to `lj.Batch`. [#22](https://github.com/elastic/go-lumber/pull/22)
```
Comment (efd6) [inline CHANGELOG.md]: Should this be 29?

File: lj/lj.go [hunk 2]
```diff
@@ -18,17 +18,28 @@
...
+	ack        chan struct{}
+	TLS        *tls.ConnectionState // TLS connection metadata. Nil for non-TLS connections.
+	SourceAddr string               // Source address of the connection.
+	Events     []interface{}
 }
 
 // NewBatch creates a new ACK-able batch.
-func NewBatch(evts []interface{}) *Batch {
-	return &Batch{evts, make(chan struct{})}
+func NewBatch(evts []interface{}, remoteAddr string, tlsState *tls.ConnectionState) *Batch {
```
Comment (efd6) [inline lj/lj.go]: Doc addition for `remoteAddr` and `tlsState`?

Also, this is a breaking change. Maybe add as `NewBatchTLS` and have `NewBatch(evts []... { return NewBatchTLS(evts, "", nil) }`.

File: lj/lj.go [hunk 3]
```diff
@@ -18,17 +18,28 @@
...
+
 // Batch is an ACK-able batch of events as has been received by lumberjack
 // server implemenentations. Batches must be ACKed, for the server
 // implementations returning an ACK to it's clients.
 type Batch struct {
-	Events []interface{}
-	ack    chan struct{}
+	ack        chan struct{}
+	TLS        *tls.ConnectionState // TLS connection metadata. Nil for non-TLS connections.
+	SourceAddr string               // Source address of the connection.
```
Comment (efd6) [inline lj/lj.go]: Rename to `RemoteAddr` for consistency with the constructor signature and `tls.Conn`?

File: server/v1/reader.go [hunk 4]
```diff
@@ -32,18 +33,26 @@ import (
...
 type reader struct {
-	in      *bufio.Reader
-	conn    net.Conn
-	timeout time.Duration
-	buf     []byte
+	conn       net.Conn
+	in         *bufio.Reader
+	tlsState   *tls.ConnectionState
+	remoteAddr string
+	buf        []byte
```
Comment (efd6) [inline server/v1/reader.go:40]: If we are confident that it's nearly always the case that the `buf` is not grown beyond 64 bytes long, maybe make it be a slice of another `[64]byte` field?

File: server/v2/reader.go [hunk 5]
```diff
@@ -32,22 +33,30 @@ import (
...
-	conn    net.Conn
-	timeout time.Duration
-	decoder jsonDecoder
-	buf     []byte
+	conn       net.Conn
+	in         *bufio.Reader
+	tlsState   *tls.ConnectionState
+	decoder    jsonDecoder
+	remoteAddr string
+	buf        []byte
```
Comment (efd6) [inline server/v2/reader.go:41]: Same comment here.

File: server/v1/reader.go (hunk 4 above)
Comment (andrewkroh) [reply server/v1/reader.go:40]: I'm not confident on this so I'm going to leave it as is. But good idea.

File: lj/lj.go (hunk 2 above)
Comment (andrewkroh) [reply lj/lj.go]: Changed to `NewBatchWithSourceMetadata`.

**Review by efd6 — APPROVED** [review]

**Pull request merged**